components/channel_layout/center_channel.jsx

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Route, Switch, Redirect} from 'react-router-dom';
+
+import ChannelView from 'components/channel_view';
+import PermalinkView from 'components/permalink_view';
+import MessageIdentifierRouter from 'components/channel_layout/message_indentifier_router';
+import ChannelStore from 'stores/channel_store';
+import TeamStore from 'stores/team_store';
```

Comment:
If we could replace these with using the redux store directly, that would be great :)

Reply:
I have never seen this before. Could you point me in the right direction on a file that does it correctly? I am not even sure I understand why it was done this way in the first place... thanks

Reply:
Basically either have anything you need from the redux store passed in through props, or if that's not desired [you can do something like this](https://github.com/mattermost/mattermost-webapp/blob/master/actions/websocket_actions.jsx#L528) where you just pull state directly out of the redux store

The long story made short is that most of the stuff in the `/stores` dir (such as UserStore, ChannelStore, etc.) is legacy flux stores that were converted to use redux underneath the covers to make the migration to redux easier. The goal is to eventually remove all uses of these stores so we can be completely on the redux pattern

Reply:
ok. So you are saying use redux connect to pass in as props, don't use the current implementation, right? I think I can do that. 👍